src/abstractblocksparsearray/abstractblocksparsematrix.jl

```diff
@@ -1 +1,22 @@
 const AbstractBlockSparseMatrix{T} = AbstractBlockSparseArray{T,2}
+
+# SVD is implemented by trying to
+# 1. Attempt to find a block-diagonal implementation by permuting
+# 2. Fallback to AbstractBlockArray implementation via BlockedArray
+function svd(
+  A::AbstractBlockSparseMatrix; full::Bool=false, alg::Algorithm=default_svd_alg(A)
+)
+  T = LinearAlgebra.eigtype(eltype(A))
+  A′ = try_to_blockdiagonal(A)
+
+  if isnothing(A′)
+    # not block-diagonal, fall back to dense case
+    Adense = eigencopy_oftype(A, T)
+    return svd!(Adense; full, alg)
+  end
+
+  # compute block-by-block and permute back
+  A″, (I, J) = A′
+  F = svd!(eigencopy_oftype(A″, T); full, alg)
+  return SVD(F.U[Block.(I), Block.(J)], F.S, F.Vt)
```

Comment:
Can `I` and `J` already be designed as lists of `Block`?

Reply:
Actually it looks like that is already the case so it seems like `Block.(...)` may not be necessary.

Reply:
I'm a bit confused by this code, I would have though it would be something like:
```julia
SVD(F.U[I, :], F.S, F.Vt[:, J])
```
i.e. `I` and `J` are block permutations of the outer spaces but maybe I have the wrong mental model for how this function is being designed.